`new Vue` is checked by component-related rules

<!--
  ❗Please don't ignore this template.❗

  If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
  The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
  Before posting the issue, please confirm that the problem you're getting
  is not related with your code editor configuration.
  To make sure it's not, run: yarn eslint src/your-file.vue
-->

**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.
<!-- If you do not read the FAQ and open an issue that is listed in the FAQ, we may silently close the issue. -->

**Tell us about your environment**

- **ESLint version:** 8.31.0
- **eslint-plugin-vue version:** 9.9.0
- **Node version:** 16.16.0
- **Operating System:** macOS

**Please show your full configuration:**
<!-- Paste content of your .eslintrc file -->
```js
module.exports = {
  root: true,
  extends: ['plugin:vue/vue3-recommended', 'eslint:recommended'],
  rules: {
    'vue/require-name-property': ['error'],
    'vue/padding-lines-in-component-definition': ['error', {
      betweenOptions: `always`,
      groupSingleLineProperties: false,
    }]
  },
  parserOptions: {
    ecmaVersion: 'latest',
  },
}
```

**What did you do?**	
<!-- Please include the actual source code causing the issue. -->
```js
import Vue from 'vue'
import App from './App.vue'
import { createPinia } from 'pinia'

const pinia = createPinia()

new Vue({
  pinia,
  ...App
}).$mount('#app')
```

**What did you expect to happen?**

Everything works fine and no error reported since `new Vue` is not listed in [docs - eslint-detect-components](https://eslint.vuejs.org/user-guide/#how-does-eslint-detect-components).

**What actually happened?**
<!--
  Please include the actual, raw output from ESLint.
  If you are only looking at the results of your editor extension, also check the CLI results.
-->

ESLint reported 2 errors:

```
  7:9  error  Required name property is not set           vue/require-name-property
  9:3  error  Expected blank line before this definition  vue/padding-lines-in-component-definition
```

**Repository to reproduce this issue**
<!--
  ❗ Please share a repository that can reproduce your issue.
  ❗ If you don't share it, we just label it as requesting a repository that can reproduce your issue.
  ❗ If you don't share it for a few days, we silently close this issue.
-->

[https://github.com/ntnyq-dev/repro-eslint-plugin-vue](https://github.com/ntnyq-dev/repro-eslint-plugin-vue)

**Related rules**

- `vue/require-name-property`
- `vue/padding-lines-in-component-definition`
- others

**Additional**

If it is expected, how can I exclude it from being checked?

--- a/docs/user-guide/index.md
+++ b/docs/user-guide/index.md
@@ -323,7 +323,7 @@ All component-related rules are applied to code that passes any of the following
 - `component()` expression
 - `defineComponent()` expression
 - `defineNuxtComponent()` expression
-- `export default {}` in `.vue`, `.jsx`, or `.tsx` file
+- `export default {}` in `.vue`, `.jsx`, or `.tsx` files
 
 However, if you want to take advantage of the rules in any of your custom objects that are Vue components, you might need to use the special comment `// @vue/component` that marks an object in the next line as a Vue component in any file, e.g.:
 

--- a/docs/user-guide/index.md
+++ b/docs/user-guide/index.md
@@ -316,6 +316,7 @@ All component-related rules are applied to code that passes any of the following
 - `Vue.component()` expression
 - `Vue.extend()` expression
 - `Vue.mixin()` expression
+- `new Vue()` expression
 - `app.component()` expression
 - `app.mixin()` expression
 - `createApp()` expression

--- a/docs/user-guide/index.md
+++ b/docs/user-guide/index.md
@@ -314,6 +314,7 @@ The `parserOptions.parser` option can also specify an object to specify multiple
 All component-related rules are applied to code that passes any of the following checks:
 
 - `Vue.component()` expression
+- `component()` expression
 - `Vue.extend()` expression
 - `Vue.mixin()` expression
 - `new Vue()` expression
@@ -321,7 +322,8 @@ All component-related rules are applied to code that passes any of the following
 - `app.mixin()` expression
 - `createApp()` expression
 - `defineComponent()` expression
-- `export default {}` in `.vue` or `.jsx` file
+- `defineNuxtComponent()` expression
+- `export default {}` in `.vue`, `.jsx`, or `.tsx` file
 
 However, if you want to take advantage of the rules in any of your custom objects that are Vue components, you might need to use the special comment `// @vue/component` that marks an object in the next line as a Vue component in any file, e.g.:
 

--- a/docs/user-guide/index.md
+++ b/docs/user-guide/index.md
@@ -314,13 +314,13 @@ The `parserOptions.parser` option can also specify an object to specify multiple
 All component-related rules are applied to code that passes any of the following checks:
 
 - `Vue.component()` expression
-- `component()` expression
 - `Vue.extend()` expression
 - `Vue.mixin()` expression
-- `new Vue()` expression
 - `app.component()` expression
 - `app.mixin()` expression
+- `new Vue()` expression
 - `createApp()` expression
+- `component()` expression
 - `defineComponent()` expression
 - `defineNuxtComponent()` expression
 - `export default {}` in `.vue`, `.jsx`, or `.tsx` file

